### PR TITLE
move process_message to its own crate

### PR DESCRIPTION
**Pull request by kevinheavey**
#### Problem
`litesvm` uses the `solana-svm` crate only for the associated function `MessageProcessor::process_message`. It would need to compile much less code if this function lived in its own crate.

#### Summary of Changes
- Move `MessageProcessor::process_message` to a new crate `solana-svm-message-processor` as a free function
- Call this free function in `MessageProcessor::process_message` so it still works as before
- Deprecate the `MessageProcessor` struct in `solana-svm` as it now does nothing
- Update the usage of `process_message` in this repo

### CODE REVIEW

File: svm-message-processor/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,680 @@
...
+    },
+    solana_svm_transaction::svm_message::SVMMessage,
+    solana_timings::{ExecuteDetailsTimings, ExecuteTimings},
+};
+
+/// Process a message.
+/// This method calls each instruction in the message over the set of loaded accounts.
+/// For each instruction it calls the program entrypoint method and verifies that the result of
+/// the call does not violate the bank's accounting rules.
+/// The accounts are committed back to the bank only if every instruction succeeds.
```
Comment (apfitzge) [inline svm-message-processor/src/lib.rs]: Let's leave any comments about comitting and banks out of this, they are a higher-level concept that really isn't relevant here (imo).

What is important here is that a `TransactionError` is returned if **any** of the instructions fail

Comment (kevinheavey) [reply svm-message-processor/src/lib.rs]: Oh I didn't write this I just copied the associated function along with its doc comment from here: https://github.com/anza-xyz/agave/blob/990f938a718d4bcc791e26002a5475a233afa4c8/svm/src/message_processor.rs#L30

Want me to change it while we're here?

Comment (apfitzge) [reply svm-message-processor/src/lib.rs]: yes please 😄

**Comment by apfitzge** [comment]
move itself looks good to me. I tagged Joe since he generally does stuff related to the svm public interface

**Review by buffalojoec — COMMENTED** [review]
Sorry @kevinheavey but we're not ready to start breaking up the SVM crate just yet. Many of the APIs are still changing, and there are a few ongoing SIMD-related changes that have to go into this crate.

When SVM moves to its own repo, and we get a better sense of what we might like the various APIs offered by those libraries to look like, we can revisit this.

**Comment by kevinheavey** [comment]
Why does pulling out a new crate hinder any of that work?

**Pull request closed without merging**